### PR TITLE
Drop support for gdb lt 9.2 #1854

### PR DESCRIPTION
**Pull request by dmur1**
https://github.com/pwndbg/pwndbg/issues/1854

### CODE REVIEW

**Comment by disconnect3d** [comment]
That's irrelevant for the PR but eh, Arch Linux changed their glibc or sth? 
image

File: pwndbg/gdblib/events.py [hunk 1]
```diff
@@ -115,15 +115,10 @@ def invoke_callbacks(self) -> None:
...
+    gdb.events.memory_changed: [],
+    gdb.events.register_changed: []
 }
 
-# GDB 7.9 and above only
-try:
-    registered[gdb.events.memory_changed] = []
-    registered[gdb.events.register_changed] = []
-except (NameError, AttributeError):
-    pass
```
Comment (disconnect3d) [inline pwndbg/gdblib/events.py:125]: Refactored this further by moving it up to the dict

File: pwndbg/gdblib/regs.py [hunk 2]
```diff
@@ -48,21 +37,15 @@ class module(ModuleType):
     def __getattr__(self, attr: str) -> int:
         attr = attr.lstrip("$")
         try:
-            # Seriously, gdb? Only accepts uint32.
-            if "eflags" in attr or "cpsr" in attr:
-                value = gdb77_get_register(attr)
```
Comment (disconnect3d) [inline pwndbg/gdblib/regs.py:53]: We removed this part: did we check if `pwndbg.gdblib.regs.eflags` work after this?

Comment (disconnect3d) [reply pwndbg/gdblib/regs.py:53]: (also random thought: `x in attr` was so wrong :D)

File: pwndbg/gdblib/regs.py [hunk 3]
```diff
@@ -48,21 +37,15 @@ class module(ModuleType):
...
-                    value += self.cs * 16
-
+            value = gdb_get_register(attr)
+            if value is None and attr.lower() == "xpsr":
+                value = gdb_get_register("xPSR")
+            size = pwndbg.gdblib.typeinfo.unsigned.get(
+                value.type.sizeof, pwndbg.gdblib.typeinfo.ulong
+            )
+            value = value.cast(size)
+            if attr.lower() == "pc" and pwndbg.gdblib.arch.current == "i8086":
```
Comment (disconnect3d) [inline pwndbg/gdblib/regs.py:47]: I really wonder if we have to do `.lower()` here at all, I think we should never have here an uppercase value ( ? )

Would need to check different archs if we have there uppercase reg names lol

File: pwndbg/gdblib/regs.py [hunk 4]
```diff
@@ -48,21 +37,15 @@ class module(ModuleType):
...
-                size = pwndbg.gdblib.typeinfo.unsigned.get(
-                    value.type.sizeof, pwndbg.gdblib.typeinfo.ulong
-                )
-                value = value.cast(size)
-                if attr.lower() == "pc" and pwndbg.gdblib.arch.current == "i8086":
-                    value += self.cs * 16
-
+            value = gdb_get_register(attr)
+            if value is None and attr.lower() == "xpsr":
+                value = gdb_get_register("xPSR")
```
Comment (disconnect3d) [inline pwndbg/gdblib/regs.py:42]: I wonder if this does not work with lowercase xpsr; this really should have a comment here

File: pwndbg/gdblib/regs.py [hunk 5]
```diff
@@ -186,13 +169,10 @@ def gsbase(self):
 
     @pwndbg.lib.cache.cache_until("stop")
     def _fs_gs_helper(self, regname: str, which):
-        """Supports fetching based on segmented addressing, a la fs:[0x30].
-        Requires ptrace'ing the child directly for GDB < 8."""
```
Comment (disconnect3d) [inline pwndbg/gdblib/regs.py:190]: we still do ptrace, but for i386, lets state that in comment

**Comment by disconnect3d** [comment]
@dmur1 I have merged this because its good enough. Can you look at my comments and resolve them in another PR?

**Comment by dmur1** [comment]
> @dmur1 I have merged this because its good enough. Can you look at my comments and resolve them in another PR?

will do this now

File: pwndbg/gdblib/regs.py (hunk 4 above)
Comment (dmur1) [reply pwndbg/gdblib/regs.py:42]: not sure what you mean with this one

**Comment by dmur1** [comment]
here you go https://github.com/pwndbg/pwndbg/pull/1881

**Pull request merged**